Utils/UpdateList.cs

```diff
@@ -16,36 +15,57 @@ public class UpdateList
 		private readonly List<int> deadFunctionIndices = new List<int>();
 
 		private bool isUpdating;
-		private int currentUpdatingIndex;
 
+		/// <summary>
+		/// Returns true if the update list contains this function
+		/// </summary>
+		/// <param name="updateFunction">The update function to test for the presence of</param>
+		/// <returns>True if the function is in the update list, false if not</returns>
+		public bool Contains(Action<float> updateFunction)
+		{
+			var index = updateFunctions.IndexOf(updateFunction);
+			return index != -1 && !deadFunctionIndices.Contains(index);
+		}
+
+		/// <summary>
+		/// Adds the given function to the update list. It must not already be added.
+		/// </summary>
+		/// <param name="updateFunction">The update function to add to the update list</param>
+		/// <exception cref="Exception">Throws an Exception if the function was already added</exception>
 		public void Add(Action<float> updateFunction)
 		{
 			var index = updateFunctions.IndexOf(updateFunction);
 			if (index >= 0)
 			{
-				// Someone wants to add it back right after removed it
-				// Possibally, Abort() then Play() in an update callback
+				// if the function exists but it's marked as dead we should, unmark it (to re add it)
 				if (deadFunctionIndices.Contains(index))
 				{
-					// We just cancel the removement
 					deadFunctionIndices.Remove(index);
 					return;
 				}
-				Debug.LogWarning("You should not add the same update function more than once into update container.");
-				return;
+
+				throw new Exception("Cannot add the given function. It is already in the list");
```

Comment:
Just repeating what I spoke to you earlier about, is there a case where this could happen when it should be expected to work? E.g. if an animation calls .Play on itself again in it's onComplete.

Reply:
In the case you describe it would have been removed, since it's removed before notifying onComplete handlers. If it's not yet been removed but marked as dead then it handles this also lines 41 to 45

Reply:
So it does!